react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentIndex: 0,
+      intervalID: null
+    };
+    this.showNextIndex = this.showNextIndex.bind(this);
+    this.showPreviousIndex = this.showPreviousIndex.bind(this);
+    this.showClicked = this.showClicked.bind(this);
+  }
+
+  componentDidMount() {
+    const intervalID = setInterval(this.showNextIndex, 1000);
+    this.setState({ intervalID });
+  }
+
+  showNextIndex() {
+    this.state.currentIndex === 4
+      ? this.setState({ currentIndex: 0 })
+      : this.setState({ currentIndex: this.state.currentIndex + 1 });
+    this.controlInterval();
+  }
+
+  showPreviousIndex() {
+    this.state.currentIndex === 0
+      ? this.setState({ currentIndex: 4 })
+      : this.setState({ currentIndex: this.state.currentIndex - 1 });
```

Comment:
Once you switch your input data to an array, you can base these conditions on its length. Again, trying to avoid making assumptions about our inputs so that new / different props don't require updating the component's code